src/app/view-all-questions/page.tsx

```diff
@@ -0,0 +1,77 @@
+'use client'
+
+import { useEffect, useState } from 'react';
+import { LoadingSpinner } from '@/components/ui/loading';
+import { type Question } from '@/types/Question';
+
+const QuestionsPage: React.FC = () => {
+  const [questions, setQuestions] = useState<Question[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [hasError, setHasError] = useState<boolean>(false);
+
+  useEffect(() => {
+    const fetchQuestions = async () => {
+      try {
+        setIsLoading(true);
+        const res = await fetch('http://localhost:8000/api/getAllQuestions');
+        if (!res.ok) {
+          throw new Error(`HTTP error! Status: ${res.status}`);
+        }
+        const data = await res.json() as Question[];
+        const sortedQuestions = data.sort(
+          (a, b) => new Date(b.created_at).getTime() - new Date(a.created_at).getTime()
+        );
+        setQuestions(sortedQuestions);
+      } catch (error) {
+        console.error('Error fetching questions:', error);
+        setHasError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    void fetchQuestions();
+  }, []);
+
+  return (
+    <div className="max-w-4xl mx-auto p-6">
+      <h1 className="text-3xl font-bold text-center text-primary mb-8 text-balance">Questions</h1>
+
+      {isLoading && (
+        <div className="flex flex-col items-center justify-center my-10">
+          <LoadingSpinner />
+          <p className="mt-4 text-muted">Loading questions...</p>
+        </div>
+      )}
+
+      {hasError && (
+        <div className="text-center my-10 text-destructive">
+          <p>Something went wrong while fetching the questions. Please try again later.</p>
+        </div>
+      )}
+
+      {!isLoading && !hasError && (
+        <ul className="space-y-6">
+          {questions.map((question) => (
+            <li key={question.question_id.toString()} className="p-6 rounded-lg shadow-md bg-card border border-border">
```

Comment:
I think it would be useful to add an onClick method here to navigate to the corresponding question.

You would just need `import { useRouter } from 'next/navigation'` at the top of the file, and  include `const router = useRouter()` at the top of your component. (see the ask-question page for an example)

The onClick would look like this:
```typescript
router.push(`/questions/${question.questionId}`)
```